openupgradelib/openupgrade_merge_records.py

```diff
@@ -431,6 +431,111 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
         )
 
 
+def _adjust_merged_values_sql(env, model_name, record_ids, target_record_id,
+                              exclude_columns, model_table, field_spec):
+    """This method deals with the values on the records to be merged +
+    the target record, performing operations that makes sense on the meaning
+    of the model.
+
+    :param field_spec: Dictionary with field names as keys and forced operation
+      to perform as values. If a field is not present here, default operation
+      will be performed.
+
+      Possible operations by field types same as _adjust_merged_values_orm.
+    """
+    if not column_exists(env.cr, model_table, 'id'):
+        return
+    env.cr.execute(
+        """
+        SELECT isc.column_name, isc.data_type, imf.ttype
+        FROM information_schema.columns isc
+        JOIN ir_model_fields imf ON (
+            imf.name = isc.column_name AND imf.model = %s)
+        WHERE table_name = %s
+        """, (model_table, model_name))
+    vals = {}
+    new_vals = {}
+    for column, column_type, field_type in env.cr.fetchall():
+        if (model_table, column) in exclude_columns:
+            continue
+        op = field_spec.get(column, False)
+        env.cr.execute(
+            sql.SQL(
+                """SELECT {column}
+                FROM {table}
+                WHERE id IN %(record_ids)s"""
+            ).format(
+                table=sql.Identifier(model_table),
+                column=sql.Identifier(column),
+            ), {'record_ids': (target_record_id,) + tuple(record_ids)}
+        )
+        _list = [x[0] for x in env.cr.fetchall()]
+        if field_type in ('char', 'text', 'html'):
```

Comment:
As this code is shared between both methods, I think you can extract it and pass by parameters the needed values to be handled, for saving code.